src/appmixer/quickbooks/accounting/NewInvoice/NewInvoice.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const { makeRequest } = require('../../commons');
+
+module.exports = {
+
+    async tick(context) {
+
+        let { changedSince } = await context.stateGet('changedSince') || {};
+
+        if (!changedSince) {
+            // On first tick, fetch only the most recent update to set changedSince
+            changedSince = new Date();
+            await context.stateSet('changedSince', { changedSince });
+        }
+
+        const options = {
+            path: `v3/company/${context.profileInfo.companyId}/cdc?entities=Invoice&changedSince=${changedSince.toISOString()}`,
+            method: 'GET'
+        };
+
+        try {
+            const { data } = await makeRequest({ context, options });
+
+            if (data.CDCResponse[0].QueryResponse[0].Invoice) {
+                // Empty response: {"CDCResponse": [{"QueryResponse": [{}]}],"time": "2024-04-18T05:12:07.655-07:00"}
+                for (const entity of data.CDCResponse[0].QueryResponse[0].Invoice) {
+                    // Compare MetaData.CreateTime with MetaData.LastUpdatedTime to determine if the entity is new or updated
+                    const isNew = entity.MetaData.CreateTime === entity.MetaData.LastUpdatedTime;
```

Comment:
this could also throws a confusing error  like "Cannot read properties of undefined"

Reply:
refactored, even though `Metadata` should be in every entity.